pkg/scheduler/core/generic_scheduler.go

```diff
@@ -434,21 +441,57 @@ func (g *genericScheduler) prioritizeNodes(
 		return nil, scoreStatus.AsError()
 	}
 
-	if klog.V(10).Enabled() {
-		for plugin, nodeScoreList := range scoresMap {
-			for _, nodeScore := range nodeScoreList {
-				klog.InfoS("Plugin scored node for pod", "pod", klog.KObj(pod), "plugin", plugin, "node", nodeScore.Name, "score", nodeScore.Score)
-			}
-		}
-	}
-
 	// Summarize all scores.
 	result := make(framework.NodeScoreList, 0, len(nodes))
+	totalPluginScores := make(map[string]int64)
```

Comment:
this is adding to the production runtime even if V(4) is not satisfied. You have to do it inside the conditional, even if it repeats code.

Reply:
+1 so that `totalPluginScores[j] += scoresMap[j][i].Score` below will not always costs.

Reply:
I moved `totalPluginScores[j] += scoresMap[j][i].Score` under a `V(4)` check, but the variable initialization doesn't work if it's put into a conditional context

Reply:
I was suggesting you do variable initialization and iteration over `scoresMap` inside the big conditional below. To keep all the changes related to logging in a single location.

Reply:
this is still unresolved

Reply:
Oh, I think I get what you mean. I moved this to `logPluginScores` with its own iteration over `scoresMap`. Is that what you meant?